apps/nextjs/src/app/[locale]/manage/tools/kubernetes/cluster-dashboard/cluster-dashboard.tsx

```diff
@@ -0,0 +1,78 @@
+"use client";
+
+import { SimpleGrid, Skeleton, Stack, Title } from "@mantine/core";
+import { nanoid } from "nanoid";
+
+import { clientApi } from "@homarr/api/client";
+import { useI18n } from "@homarr/translation/client";
+
+import KubernetesErrorPage from "~/app/[locale]/manage/tools/kubernetes/cluster-dashboard/error";
+import { HeaderCard } from "~/app/[locale]/manage/tools/kubernetes/cluster-dashboard/header-card/header-card";
+import { ResourceGauge } from "~/app/[locale]/manage/tools/kubernetes/cluster-dashboard/resource-gauge/resource-gauge";
+import { ResourceTile } from "~/app/[locale]/manage/tools/kubernetes/cluster-dashboard/resource-tile/resource-tile";
+
+export function ClusterDashboard() {
+  const t = useI18n();
+
+  const {
+    data: clusterData,
+    isLoading: isClusterLoading,
+    isError: isClusterError,
+  } = clientApi.kubernetes.getCluster.useQuery();
+
+  const {
+    data: resourceCountsData,
+    isLoading: isResourceCountsLoading,
+    isError: isResourceCountsError,
+  } = clientApi.kubernetes.getClusterResourceCounts.useQuery();
+
+  if (isClusterError || isResourceCountsError) {
+    return (
+      <Stack bg="var(--mantine-color-body)">
+        <Title>{t("kubernetes.cluster.title")}</Title>
+        <KubernetesErrorPage />
+      </Stack>
+    );
+  }
+
+  return (
+    <Stack bg="var(--mantine-color-body)">
+      <Title>{t("kubernetes.cluster.title")}</Title>
+      <SimpleGrid cols={{ xs: 1, sm: 2, md: 3 }}>
+        {isClusterLoading ? (
+          Array.from({ length: 3 }).map(() => <Skeleton key={nanoid()} height={65} />)
+        ) : (
+          <>
+            <HeaderCard headerType={"providers"} value={clusterData ? clusterData.providers : ""} />
+            <HeaderCard headerType={"version"} value={clusterData ? clusterData.kubernetesVersion : ""} />
+            <HeaderCard headerType={"architecture"} value={clusterData ? clusterData.architecture : ""} />
+          </>
+        )}
+      </SimpleGrid>
+
+      <Title>{t("kubernetes.cluster.capacity.title")}</Title>
+
+      <SimpleGrid cols={{ xs: 1, sm: 2, md: 3 }}>
+        {isClusterLoading
+          ? Array.from({ length: 3 }).map(() => <Skeleton key={nanoid()} height={200} />)
```

Comment:
Pretty sure you shouldn't use indexes or random IDs for this...
@Meierschlumpf can you confirm?

Reply:
Indexes are fine in such cases as the amount of items does not change and so react does not need to understand position changes of certain elements. Random ids are not necessary